homeassistant/helpers/update_coordinator.py

```diff
@@ -104,8 +104,14 @@ def _schedule_refresh(self) -> None:
         """
         await self._debounced_refresh.async_call()
 
-    async def async_refresh(self) -> None:
+    async def async_update_data(self) -> Optional[Any]:
         """Update data."""
+        if self.update_method is None:
+            raise UpdateFailed("Update method not implemented")
```

Comment:
This should not be an `UpdateFailed`. Raise `NotImplementedError`.

Reply:
Currently `async_refresh` catches `Exception`, so nothing will bubble up.

Considering your request of using `NotImplementedError` I'd like to suggest to re-raise that exception in the `async_refresh` method.

Reply:
I want this class to crash hard if this method doesn't exist. It's an implementation bug. Can we re-raise this in refresh too ?

Reply:
That was my proposal indeed...

I will make sure it crashes.

Reply:
Done.